Build failure on Ubuntu 24.04

Been using s3ql without issue for years. Moving to new server, and attempted to build/install on Ubuntu 24.04, but results in fatal errors during build, all seem to be related to sqlite3ext.cpp: 
```
s3ql-5.2.2$ python3 setup.py build_ext --inplace -f
running build_ext
building 's3ql.sqlite3ext' extension
x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/usr/include/python3.12 -c src/s3ql/sqlite3ext.cpp -o build/temp.linux-x86_64-cpython-312/src/s3ql/sqlite3ext.o -Wall -Wextra -Wconversion -Wsign-compare
src/s3ql/sqlite3ext.cpp: In function ‘PyObject* __pyx_pf_4s3ql_10sqlite3ext_track_writes(PyObject*, PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.12/dictobject.h:90,
                 from /usr/include/python3.12/Python.h:64,
                 from src/s3ql/sqlite3ext.cpp:25:
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: At global scope:
src/s3ql/sqlite3ext.cpp:2679:1: warning: missing initializer for member ‘_typeobject::tp_watched’ [-Wmissing-field-initializers]
 2679 | };
      | ^
src/s3ql/sqlite3ext.cpp: In function ‘void __Pyx_Raise(PyObject*, PyObject*, PyObject*, PyObject*)’:
src/s3ql/sqlite3ext.cpp:3636:36: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
 3636 |         PyObject* tmp_tb = tstate->curexc_traceback;
      |                                    ^~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3639:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
 3639 |             tstate->curexc_traceback = tb;
      |                     ^~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘uint64_t __Pyx_get_tp_dict_version(PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘uint64_t __Pyx_get_object_dict_version(PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_object_dict_version_matches(PyObject*, uint64_t, uint64_t)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘PyObject* __Pyx__GetModuleGlobalName(PyObject*, uint64_t*, PyObject**)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_CLineForTraceback(PyThreadState*, int)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4137:69: warning: unused parameter ‘tstate’ [-Wunused-parameter]
 4137 |  int __Pyx_CLineForTraceback(CYTHON_NCP_UNUSED PyThreadState *tstate, int c_line) {
      |                                                ~~~~~~~~~~~~~~~^~~~~~

src/s3ql/sqlite3ext.cpp: In function ‘size_t __Pyx_PyInt_As_size_t(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4413:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4413 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4468:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4468 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘long int __Pyx_PyInt_As_long(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4647:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4647 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4702:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4702 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_PyInt_As_int(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4843:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4843 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4898:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4898 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘Py_ssize_t __Pyx_PyIndex_AsSsize_t(PyObject*)’:
src/s3ql/sqlite3ext.cpp:5349:47: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 5349 |     const digit* digits = ((PyLongObject*)b)->ob_digit;
      |                                               ^~~~~~~~
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
drgrumpy@vs02:~/downloads/s3ql/s3ql-5.2.2$ python3 setup.py build_ext --inplace -f
running build_ext
building 's3ql.sqlite3ext' extension
x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/usr/include/python3.12 -c src/s3ql/sqlite3ext.cpp -o build/temp.linux-x86_64-cpython-312/src/s3ql/sqlite3ext.o -Wall -Wextra -Wconversion -Wsign-compare
cc1plus: fatal error: src/s3ql/sqlite3ext.cpp: No such file or directory
compilation terminated.
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
drgrumpy@vs02:~/downloads/s3ql/s3ql-5.2.2$ python3 setup.py build_ext --inplace -f
running build_ext
building 's3ql.sqlite3ext' extension
x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/usr/include/python3.12 -c src/s3ql/sqlite3ext.cpp -o build/temp.linux-x86_64-cpython-312/src/s3ql/sqlite3ext.o -Wall -Wextra -Wconversion -Wsign-compare
src/s3ql/sqlite3ext.cpp: In function ‘PyObject* __pyx_pf_4s3ql_10sqlite3ext_track_writes(PyObject*, PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.12/dictobject.h:90,
                 from /usr/include/python3.12/Python.h:64,
                 from src/s3ql/sqlite3ext.cpp:25:
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1235:43: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1235 |     (var) = (likely(__pyx_dict_version == __PYX_GET_DICT_VERSION(__pyx_d))) ?\
      |                                           ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1971:3: note: in expansion of macro ‘__Pyx_GetModuleGlobalName’
 1971 |   __Pyx_GetModuleGlobalName(__pyx_t_2, __pyx_n_s_os); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 49, __pyx_L1_error)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: At global scope:
src/s3ql/sqlite3ext.cpp:2679:1: warning: missing initializer for member ‘_typeobject::tp_watched’ [-Wmissing-field-initializers]
 2679 | };
      | ^
src/s3ql/sqlite3ext.cpp: In function ‘void __Pyx_Raise(PyObject*, PyObject*, PyObject*, PyObject*)’:
src/s3ql/sqlite3ext.cpp:3636:36: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
 3636 |         PyObject* tmp_tb = tstate->curexc_traceback;
      |                                    ^~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3639:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
 3639 |             tstate->curexc_traceback = tb;
      |                     ^~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘uint64_t __Pyx_get_tp_dict_version(PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3675:27: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3675 |     return likely(dict) ? __PYX_GET_DICT_VERSION(dict) : 0;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘uint64_t __Pyx_get_object_dict_version(PyObject*)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:3687:36: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3687 |     return (dictptr && *dictptr) ? __PYX_GET_DICT_VERSION(*dictptr) : 0;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_object_dict_version_matches(PyObject*, uint64_t, uint64_t)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:987:43: note: in definition of macro ‘unlikely’
  987 |   #define unlikely(x) __builtin_expect(!!(x), 0)
      |                                           ^
src/s3ql/sqlite3ext.cpp:3691:56: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 3691 |     if (unlikely(!dict) || unlikely(tp_dict_version != __PYX_GET_DICT_VERSION(dict)))
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘PyObject* __Pyx__GetModuleGlobalName(PyObject*, uint64_t*, PyObject**)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1209:21: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1209 |     (version_var) = __PYX_GET_DICT_VERSION(dict);\
      |                     ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:3708:5: note: in expansion of macro ‘__PYX_UPDATE_DICT_CACHE’
 3708 |     __PYX_UPDATE_DICT_CACHE(__pyx_d, result, *dict_cached_value, *dict_version)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_CLineForTraceback(PyThreadState*, int)’:
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:986:43: note: in definition of macro ‘likely’
  986 |   #define likely(x)   __builtin_expect(!!(x), 1)
      |                                           ^
src/s3ql/sqlite3ext.cpp:1214:16: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1214 |     if (likely(__PYX_GET_DICT_VERSION(DICT) == __pyx_dict_version)) {\
      |                ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:1207:65: warning: ‘PyDictObject::ma_version_tag’ is deprecated [-Wdeprecated-declarations]
 1207 | efine __PYX_GET_DICT_VERSION(dict)  (((PyDictObject*)(dict))->ma_version_tag)
      |                                                               ^~~~~~~~~~~~~~

src/s3ql/sqlite3ext.cpp:1218:30: note: in expansion of macro ‘__PYX_GET_DICT_VERSION’
 1218 |         __pyx_dict_version = __PYX_GET_DICT_VERSION(DICT);\
      |                              ^~~~~~~~~~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4150:9: note: in expansion of macro ‘__PYX_PY_DICT_LOOKUP_IF_MODIFIED’
 4150 |         __PYX_PY_DICT_LOOKUP_IF_MODIFIED(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.12/cpython/dictobject.h:22:34: note: declared here
   22 |     Py_DEPRECATED(3.12) uint64_t ma_version_tag;
      |                                  ^~~~~~~~~~~~~~
src/s3ql/sqlite3ext.cpp:4137:69: warning: unused parameter ‘tstate’ [-Wunused-parameter]
 4137 |  int __Pyx_CLineForTraceback(CYTHON_NCP_UNUSED PyThreadState *tstate, int c_line) {
      |                                                ~~~~~~~~~~~~~~~^~~~~~

src/s3ql/sqlite3ext.cpp: In function ‘size_t __Pyx_PyInt_As_size_t(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4413:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4413 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4468:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4468 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘long int __Pyx_PyInt_As_long(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4647:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4647 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4702:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4702 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘int __Pyx_PyInt_As_int(PyObject*)’:
src/s3ql/sqlite3ext.cpp:4843:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4843 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp:4898:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 4898 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                       ^~~~~~~~
src/s3ql/sqlite3ext.cpp: In function ‘Py_ssize_t __Pyx_PyIndex_AsSsize_t(PyObject*)’:
src/s3ql/sqlite3ext.cpp:5349:47: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
 5349 |     const digit* digits = ((PyLongObject*)b)->ob_digit;
      |                                               ^~~~~~~~
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1

```
Python version is 3.12.3
As far as I can tell, I've got all dependencies installed
Previously I successfully installed 5.1.3 on Ubuntu 22.04 earlier this year, and followed same notes. 

Any help appreciated

--- a/ChangeLog.rst
+++ b/ChangeLog.rst
@@ -1,3 +1,9 @@
+UNRELEASED
+=======================
+
+* Make release tarballs compatible with Python 3.12 by using a newer Cython version
+  to generate the necessary C++ extension.
+
 S3QL 5.2.2 (2024-09-01)
 =======================
 

--- a/make_release.sh
+++ b/make_release.sh
@@ -14,6 +14,12 @@ echo "Creating release tarball for ${TAG}..."
 
 git checkout -q "${TAG}"
 
+# check if we have a recent enough Cython version so that the release tarball is compatible with Python 3.12
+if ! python3 -c 'import sys; from Cython.Compiler.Version import version as cython_version; sys.exit(1) if tuple(map(int, (cython_version.split(".")[0:3]))) < (3, 0, 0) else sys.exit(0)'; then
+  printf "You need to install Cython >= 3.0.0. You have version %s installed\n" "$(python3 -c 'from Cython.Compiler.Version import version as cython_version; print(cython_version)')"
+  exit 1
+fi
+
 python3 setup.py build_cython build_ext --inplace
 ./build_docs.sh
 (cd doc/pdf && latexmk)

--- a/tests/Dockerfile
+++ b/tests/Dockerfile
@@ -2,7 +2,7 @@
 # It's tailored for running tests and building documentation.
 
 FROM ubuntu:jammy AS build
-ENV DEBIAN_FRONTEND noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update \
  && apt-get dist-upgrade -y \

